File: 2024/problem_list_10/Java/leetcode/src/test/java/org/renerod1/leetcode/problem_list_10/AppTest.java
package org.renerod1.leetcode.problem_list_10;

import static org.junit.Assert.assertEquals;

import org.junit.Test;

public class AppTest {

    App s = new App();

    @Test
    public void case1() {
        assertEquals(false, s.isMatch("aa", "a"));
    }

    @Test
    public void case2() {
        assertEquals(true, s.isMatch("aa", "a*"));
    }

    @Test
    public void case3() {
        assertEquals(true, s.isMatch("ab", ".*"));
    }

    @Test
    public void case4() {
        assertEquals(true, s.isMatch("abc", "a***abc"));
    }
}

File: 2024/problem_list_10/Kotlin/leetcode/src/test/kotlin/org/renerod1/leetcode/problem_list_10/AppTest.kt
package org.renerod1.leetcode.problem_list_10

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class AppTest {

    private lateinit var app: App

    @BeforeEach
    fun configureApp() {
        app = App()
    } 

    @Test
    fun case1() {
        assertThat(app.isMatch("aa", "a")).isEqualTo(false)
    }

    @Test
    fun case2() {
        assertThat(app.isMatch("aa", "a*")).isEqualTo(true)
    }

    @Test
    fun case3() {
        assertThat(app.isMatch("ab", ".*")).isEqualTo(true)
    }

    @Test
    fun case4() {
        assertThat(app.isMatch("abc", "a***abc")).isEqualTo(true)
    }
}